core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts

```diff
@@ -1,14 +1,142 @@
+import { API_EXPERIMENTS_URL } from './constants';
+import { AssignedExperiments, IsUserIncludedApiResponse, SdkExperimentConfig } from './models';
+import { persistData } from './persistence/persistence';
+import { dotLogger } from './utils/utils';
+
 /**
- * Represents an SDK Experiment.
- * @class
+ * `SdkExperiments` is a Typescript class to handles all operations related to fetching, storing, parsing, and navigating
+ * data for Experiments (A/B Testing).
+ *
+ * It requires a configuration object for instantiation, please instance it using the method `getInstance` sending
+ * an object with `api-key`, `server` and `debug`.
+ *
+ * Here's an example of how you can instantiate SdkExperiments class:
+ * @example
+ * ```typescript
+ * const instance = SdkExperiments.getInstance({
+ *   server: "yourServerUrl",
+ *   "api-key": "yourApiKey"
+ * });
+ * ```
+ *
+ * @export
+ * @class SdkExperiments
+ *
  */
-export class DotSdkExperiment {
+export class SdkExperiments {
+    // TODO: make this class the entrypoint of the library
+    private static instance: SdkExperiments;
+
+    constructor(private config: SdkExperimentConfig) {
+        if (!this.config['server']) {
+            throw new Error('`server` must be provided and should not be empty!');
+        }
+
+        if (!this.config['api-key']) {
+            throw new Error('`api-key` must be provided and should not be empty!');
+        }
+
+        dotLogger(
+            `SDK instanced with ${JSON.stringify(config)} configuration`,
+            this.getIsDebugActive()
+        );
+
+        // TODO: Steps
+        // - 1. Fetching and storing data ✅
+        // - 2. Parsing data for backend use
+        // - 3. Detecting and navigate to the appropriate variant
+        // - 4. Sending events to Analytics Cloud
+    }
+
+    /**
+     * Retrieves instance of SdkExperiments class if doesn't exist create a new one.
+     * If the instance does not exist, it creates a new instance with the provided configuration and calls the `getExperimentData` method.
+     *
+     * @param {SdkExperimentConfig} config - The configuration object for initializing the SdkExperiments instance.
+     * @return {SdkExperiments} - The instance of the SdkExperiments class.
+     */
+    public static getInstance(config: SdkExperimentConfig): SdkExperiments {
```

Comment:
Should this `config` be optional

Reply:
If the instance was previously created, is it necessary to continue sending the config?

Reply:
The objective is not to have multiple instances of the same class ([Creational Singleton pattern](https://github.com/dotCMS/core/pull/27704/files)). You access the method directly because it is a static method, you don't need to create an instance using new.

Reply:
That's why I'm asking @oidacra. Let's say:

```ts
// File 1
import SdkExperiments from "./sdk-experiments";

const instance = SdkExperiments.getInstance(Config);
```

```ts
// File 2
import SdkExperiments from "./sdk-experiments";

const instance = SdkExperiments.getInstance(Config); // The Config shouldn't be needed. The instance was previously created.
```

Reply:
changed. Let's see how this will evolve.